crates/oxc_linter/src/rules/unicorn/consistent_function_scoping.rs

```diff
@@ -0,0 +1,1154 @@
+            "
+						const doFoo = function(foo) {
+							const doBar = function(bar) {
+								return foo + bar;
+							};
+							return foo;
+						};
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo(foo) {
+							const doBar = function(bar) {
+								return foo + bar;
+							};
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo(foo) {
+							function doBar(bar) {
+								return foo + bar;
+							}
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo(foo = 'foo') {
+							function doBar(bar) {
+								return foo + bar;
+							}
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo() {
+							const foo = 'foo';
+							function doBar(bar) {
+								return foo + bar;
+							}
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo(foo) {
+							function doBar(bar) {
+								function doZaz(zaz) {
+									return foo + bar + zaz;
+								}
+								return bar;
+							}
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						for (let foo = 0; foo < 1; foo++) {
+							function doBar(bar) {
+								return bar + foo;
+							}
+						}
+					",
+            None,
+        ),
+        (
+            "
+						let foo = 0;
+						function doFoo() {
+							foo = 1;
+							function doBar(bar) {
+								return foo + bar;
+							}
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo = foo => {
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo =
+							foo =>
+							bar =>
+							foo + bar;
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo = () => {
+							return bar => bar;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo() {
+							return bar => bar;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo = foo => {
+							const doBar = bar => {
+								return foo + bar;
+							}
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo() {
+							{
+								const foo = 'foo';
+								function doBar(bar) {
+									return bar + foo;
+								}
+							}
+						}
+					",
+            None,
+        ),
+        // eslint-plugin-unicorn allows this
+        // but i think we should report it
+        //      (
+        //          "
+        //  					function doFoo(foo) {
+        //  						{
+        //  							function doBar(bar) {
+        //  								return bar;
+        //  							}
+        //  						}
+        //  						return foo;
+        //  					}
+        //  				",
+        //          None,
+        //      ),
```

Comment:
> This rule does not detect or remove extraneous code blocks inside of functions:

^^ from https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/consistent-function-scoping.md

i think we should support this case - thoughts?